java/java.source.base/src/org/netbeans/api/java/source/TreeUtilities.java

```diff
@@ -1355,7 +1355,11 @@ public StatementTree getBreakContinueTarget(TreePath breakOrContinue) throws Ill
         
         switch (leaf.getKind()) {
             case BREAK:
-                return (StatementTree) ((JCTree.JCBreak) leaf).target;
+                Tree breakTarget = (Tree) ((JCTree.JCBreak) leaf).target;
+                if (breakTarget == null || !(breakTarget instanceof StatementTree)) {
+                    return null;
```

Comment:
So.... this may be OK (or the only possibility) for this method, but as this is inevitably behavioraly incompatible, what about the clients? As the report is about the NPE check hint, that hint will be at least slightly broken in presence of switch expressions, won't it be?

For compatibility, I don't think we can avoid the behavioral incompatibility here, but then we could:
a) add a new method that would return Tree and deprecate this one
b) change the return type to Tree here, which is inevitably source incompatible. Binary compatibility might be preservable through bytecode patching.

Reply:
Thanks for the review, Jan. I see your point. It would indeed be better if I added a new method with return type as Tree. 

I have done that in the latest commit. I am only calling the new method whenever we are trying to visit break, because, while visiting break, there is always a possibility that we might hit a break that is returning a value. In most of the places, using a Tree rather than a StatementTree did not raise any concerns, except in MarkOccurencesHighlighterBase.java, where I think I have made the necessary changes.